src/app/content/components/Page/contentLinkHandler.ts

```diff
@@ -96,73 +136,98 @@ export const reduceReferences = (
 
 export const contentLinkHandler = (
   anchor: HTMLAnchorElement,
   getProps: () => ContentLinkProp,
   services: AppServices & MiddlewareAPI
 ) => async(e: MouseEvent) => {
-    const {
-      references,
-      navigate,
-      book,
-      page,
-      currentPath,
-      hasUnsavedHighlight,
-    } = getProps();
-    const href = anchor.getAttribute('href');
-    const target = anchor.getAttribute('target');
-
-    if (!href || !book || !page || target === '_blank' || isClickWithModifierKeys(e)) {
-      return;
-    }
+  const {
+    references,
+    navigate,
+    book,
+    page,
+    currentPath,
+    hasUnsavedHighlight,
+    persistentQueryParams,
+  } = getProps();
+  const href = anchor.getAttribute('href');
+  const target = anchor.getAttribute('target');
+
+  if (
+    !href ||
+    !book ||
+    !page ||
+    target === '_blank' ||
+    isClickWithModifierKeys(e)
+  ) {
+    return;
+  }
 
-    const base = new URL(assertWindow().location.href);
-    base.hash = '';
-    base.search = '';
+  const base = new URL(assertWindow().location.href);
+  base.hash = '';
+  base.search = '';
 
-    const {hash, search, pathname} = new URL(href, base.href);
-    const reference = references.find(isPathReferenceForBook(pathname, book));
-    const searchString = search.substring(1);
+  const { hash, search, pathname } = new URL(href, base.href);
+  const reference = references.find(isPathReferenceForBook(pathname, book));
 
-    if ((!reference && !(pathname === currentPath && hash))) {
-      return;
-    }
+  if (!reference && !(pathname === currentPath && hash)) {
+    return;
+  }
 
-    e.preventDefault();
+  e.preventDefault();
 
-    if (isHtmlElementWithHighlight(e.target)) {
-      if (!e.target.matches('.focus') && !isDoubleClick(href)) {
-        return;
-      }
-      e.stopPropagation();
-    }
-
-    if (hasUnsavedHighlight && !await showConfirmation(services)) {
+  if (isHtmlElementWithHighlight(e.target)) {
+    if (!e.target.matches('.focus') && !isDoubleClick(href)) {
       return;
     }
+    e.stopPropagation();
+  }
 
-    if (reference && !isPageReferenceError(reference)) {
-      // defer to allow other handlers to execute before nav happens
-      defer(() => navigate({
-        params: reference.params,
-        route: content,
-        state: {},
-      }, {hash, search: searchString}));
-    } else {
-      // defer to allow other handlers to execute before nav happens
-      defer(() => navigate({
-        params: getBookPageUrlAndParams(book, page).params,
-        route: content,
-        state: {},
-      }, {hash, search: searchString}));
-    }
-  };
+  if (hasUnsavedHighlight && !(await showConfirmation(services))) {
+    return;
+  }
+
+  const additionalQueryString = new URLSearchParams(
+    persistentQueryParams as Record<string, string>
+  ).toString();
+  const extendedSearchString = (search + '&' + additionalQueryString).substring(
```

Comment:
You might have to do something when `additionalQueryString` is empty so there's not a `&` at the end?

Reply:
queryString.parse cleans this up